tests/service/controllers/utils/test_project_clone.py

```diff
@@ -72,3 +75,34 @@ def test_service_user_project_clone(svc_client_cache):
         projects = [project.project_id for project in cache.get_projects(user)]
         assert project_one.project_id in projects
         assert project_two.project_id in projects
+
+
+@pytest.mark.service
+@pytest.mark.integration
+@retry_failed
+def test_service_user_non_existing_project_clone(svc_client_cache, it_remote_repo_url):
```

Comment:
I've tried this test without your fix and it still passes. Can you please re-check to make sure that it's a failing test?

Reply:
You are right. For some reason, when running the tests, the folder is cleaned up properly even without the fix. When the service is live (or when running it locally through telepresence) the folder remains. I tried to find the reason, but I couldn't identify the difference.
Do you have any idea where I could look? Or should I just remove this extra test since it's meaningless?

Reply:
I've tried with `payload["git_url"] = "https://dev.renku.ch/gitlab/renku-python-integration-tests/non-existing-project"` instead of `payload["git_url"] = "https://renkulabnonexistingwebsite.io"` and the test fails without your fix. I guess if it the hostname doesn't exist then clone fails in a different manner compared to when only the project doesn't exist.

Reply:
Ouch, I thought I covered that case in my trials, but apparently that was not the case :facepalm: 
Thanks for taking the time to try it! I've fixed the test to reflect this.